pkg/provider/aws/action/openshift-snc/openshift-snc.go

```diff
@@ -106,10 +105,9 @@ func Create(mCtxArgs *mc.ContextArgs, args *OpenshiftSNCArgs) (_ *OpenshiftSncRe
 	if err != nil {
 		return nil, err
 	}
-	// Manage AMI offering / replication
+	// check if AMI exists
 	amiName := amiName(&args.Version, &args.Arch)
-	if err = manageAMIReplication(mCtx, &args.Prefix,
-		&amiName, r.allocationData.Region, &args.Arch); err != nil {
+	if err = checkAMIExists(&amiName, r.allocationData.Region, &args.Arch); err != nil {
```

Comment:
mmm may it would be interesting get the AMI info, so we can set it on the request struct and do not need to search for it again on the deploy func?

Reply:
so to get the AMI (`ec2.LookupAMIResult`) we need the `pulumi.Context` which is only available in the deploy func

the `checkAMIExists` function is using the aws SDK and not using pulumi aws provider functions, so it doesn't need the `pulumi.Context`

Reply:
mm but what do we need? AMI ID or ARN...you can use it as regultar string in the request and then use pulumi.String(string) no?

Reply:
we need the type `ec2.LookupAMIResult` in the [`ComputeRequest`](https://github.com/redhat-developer/mapt/blob/6c86268697271a65e91482cc244dd3fde309c680/pkg/provider/aws/modules/ec2/compute/compute.go#L50) struct

currently in the `deploy` func there's call to [`GetAMIbyName`](https://github.com/redhat-developer/mapt/blob/6c86268697271a65e91482cc244dd3fde309c680/pkg/provider/aws/action/openshift-snc/openshift-snc.go#L173) and this returns the `ec2.LookupAMIResult` which is then set on the `computeRequest` struct